### PR TITLE
Implement upload function for metadata’s device profile client

### PR DESCRIPTION
**Pull request by weichou1229**
https://github.com/edgexfoundry/edgex-go/issues/572
Implement upload, uploadfile function according to 
https://github.com/edgexfoundry/edgex-go/blob/master/api/raml/core-metadata.raml#L763 and https://github.com/edgexfoundry/edgex-go/blob/master/api/raml/core-metadata.raml#L799 .

### CODE REVIEW

File: pkg/clients/metadata/device_profile.go [hunk 1]
```diff
@@ -22,6 +22,10 @@ import (
 	"github.com/edgexfoundry/edgex-go/pkg/clients"
 	"github.com/edgexfoundry/edgex-go/pkg/clients/types"
 	"github.com/edgexfoundry/edgex-go/pkg/models"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"path/filepath"
```
Comment (tsconn23) [inline pkg/clients/metadata/device_profile.go]: @weichou1229 Please put this block of imports above the github.com import list and alphabetize. Rebase your feature branch to clear the "out-of-date" message. After that, should be good.

Comment (drasko) [reply pkg/clients/metadata/device_profile.go]: `go fmt` should do this for you, please set-up your editor to use it.

File: pkg/clients/metadata/device_profile.go [hunk 2]
```diff
@@ -270,3 +276,82 @@ func (dpc *DeviceProfileRestClient) decodeDeviceProfile(resp *http.Response) (mo
...
+	resp, err := makeRequest(req)
+	if err != nil {
+		return "", err
+	}
+	if resp == nil {
+		return "", ErrResponseNil
+	}
+	defer resp.Body.Close()
+
+	// Get the body
```
Comment (drasko) [inline pkg/clients/metadata/device_profile.go]: Drop this comment, it's evident from the code.

File: pkg/clients/metadata/device_profile.go [hunk 3]
```diff
@@ -270,3 +276,82 @@ func (dpc *DeviceProfileRestClient) decodeDeviceProfile(resp *http.Response) (mo
...
+	bodyBytes, err := getBody(resp)
+	if err != nil {
+		return "", err
+	}
+	bodyString := string(bodyBytes)
+
+	// Check the response code
```
Comment (drasko) [inline pkg/clients/metadata/device_profile.go]: No need for this comment.

File: pkg/clients/metadata/device_profile.go [hunk 4]
```diff
@@ -270,3 +276,82 @@ func (dpc *DeviceProfileRestClient) decodeDeviceProfile(resp *http.Response) (mo
...
+	if resp.StatusCode != http.StatusOK {
+		return "", types.NewErrServiceClient(resp.StatusCode, bodyBytes)
+	}
+
+	return bodyString, nil
```
Comment (drasko) [inline pkg/clients/metadata/device_profile.go]: return immediately `string(bodyBytes)`, no need for intermediary `bodyString`

File: pkg/clients/metadata/device_profile.go [hunk 5]
```diff
@@ -270,3 +276,82 @@ func (dpc *DeviceProfileRestClient) decodeDeviceProfile(resp *http.Response) (mo
...
+		return "", ErrResponseNil
+	}
+	defer resp.Body.Close()
+
+	// Get the body
+	bodyBytes, err := getBody(resp)
+	if err != nil {
+		return "", err
+	}
+	bodyString := string(bodyBytes)
```
Comment (drasko) [inline pkg/clients/metadata/device_profile.go]: No need for thhis one, you can return directly `string(bodyBytes)`

**Comment by weichou1229** [comment]
@tsconn23 @drasko I have resolved requested changes, thanks.

**Comment by weichou1229** [comment]
@drasko Thanks for the correction. I have updated the code.

**Comment by cloudxxx8** [comment]
@drasko please help approve this PR if you don't have any more change request.  Thanks.

**Comment by feclare** [comment]
@cloudxxx8 Not blocking this PR, but there should be tests when adding new code.

**Pull request merged**